lib/dal/src/attribute/value.rs

```diff
@@ -549,93 +550,121 @@ impl AttributeValue {
-                let destination_component = Component::get_by_id(ctx, destination_component_id)
-                    .await
-                    .map_err(Box::new)?;
-                let source_component = Component::get_by_id(ctx, expected_source_component_id)
-                    .await
-                    .map_err(Box::new)?;
-                if source_component.to_delete() && !destination_component.to_delete() {
-                    continue;
-                }
-
-                let func_arg_id =
-                    AttributePrototypeArgument::func_argument_id_by_id(ctx, apa_id).await?;
-                let func_arg_name = ctx
-                    .workspace_snapshot()?
-                    .get_node_weight_by_id(func_arg_id)
-                    .await?
-                    .get_func_argument_node_weight()?
-                    .name()
-                    .to_owned();
-
-                let values_for_arg =
-                    match AttributePrototypeArgument::value_source_by_id(ctx, apa_id)
-                        .await?
-                        .ok_or(
-                            AttributeValueError::AttributePrototypeArgumentMissingValueSource(
-                                apa_id,
-                            ),
-                        )? {
-                        ValueSource::StaticArgumentValue(static_argument_value_id) => {
-                            vec![
-                                StaticArgumentValue::get_by_id(ctx, static_argument_value_id)
-                                    .await?
-                                    .value,
-                            ]
+        if apa_ids.is_empty() {
+            if let Some(implicit_input) =
+                Self::get_inferred_input_value(ctx, attribute_value_id).await?
+            {
+                let input_func = AttributePrototype::func_id(ctx, prototype_id).await?;
+
+                match Func::get_by_id(ctx, input_func).await? {
+                    Some(func) => {
+                        if func.name != "si:identity" {
+                            None
+                        } else {
+                            match FuncArgument::list_for_func(ctx, input_func).await?.pop() {
+                                Some(id) => func_binding_args.insert(id.name, vec![implicit_input]),
+                                None => None,
+                            }
                         }
-                        other_source => {
-                            let mut values = vec![];
+                    }
+                    None => None,
+                };
+            };
+        } else {
+            for apa_id in apa_ids {
+                let apa = AttributePrototypeArgument::get_by_id(ctx, apa_id).await?;
+                let expected_source_component_id = apa
+                    .targets()
+                    .map(|targets| targets.source_component_id)
+                    .unwrap_or(destination_component_id);
```

Comment:
I'm a bit confused at what's happening here: what (and why) are we falling back to if we don't have targets?

Reply:
This was as it was, so without loading this context in, I'm not sure.